CONTRIBUTING.md

```diff
@@ -17,3 +17,69 @@ Glint is a family of packages which all live in this repo as a Yarn workspace. T
+
+## How to run glint-language-server locally in debug mode?
+
+There are a few VSCode Launch Configurations within `./vscode/launch.json` that are handy for debugging:
+
+- Both will enable the TS/JS debugger on both the language server and the client-side VSCode extension code, meaning the debug breakpoints will pause execution to allow you to debug things like text completions
+- Debug Extension (Glint + TS)
+  - This spins up a VSCode window with both Glint and the built-in TS language server running.
+  - In this mode, both language servers will provide duplicate completions and suggestions, which can be useful for testing out feature parity between Glint and TS
+- Debug Extension (Glint Only)
+  - This is useful for testing out the "takeover" mode of running Glint, where Glint is responsible for providing all of the language features (debugging, diagnostics, etc); this is the ideal way to run Glint, but at the time of writing we have not yet achieved feature parity with built-in TS
+- By default these extensions will launch the VSCode Extension Host in the `test-packages` subfolder, which have Ember and Glimmerx apps that you can do some basic testing on
+- _TIP_: you can open any workspace with the Extension Host, meaning you can even debug the language server with breakpoints on a totally separate Ember repo, for example.
+- _NOTE_: debugging takes place within the `glint` workspace, i.e. if you are debugging completions, you'd trigger a completion within the Extension Host, and the breakpoint would pause within the Glint workspace VSCode instance.
+
+These launch configurations can be run via the Run and Debug tab in VSCode.
+
+## How to test out the VSCode extension locally?
+
+Firstly, there is an entire suite of integration tests that will spin up instances of VSCode and test out the VSCode Glint extension. These can be run from the `packages/vscode` directory via:
+
+```
+yarn run test
+```
+
+Secondly, the Launch Configurations described above (I believe) will run your client-side extension code in debug mode, along with the language server.
+
+## How to see the transformed Intermediate Representation (IR) code of a template
+
+Glint transforms template code into TypeScript code so that it can be type-checked by the vanilla TS compiler. Transformed IR code looks something like:
+
+```ts
+static {
+({} as typeof import("@glint/environment-ember-loose/-private/dsl")
+       .templateForBackingValue(this, function(𝚪, χ: typeof import("@glint/environment-ember-loose/-private/dsl")) {
+  {
+    const 𝛄 = χ.emitComponent(χ.resolve(χ.Globals["FooComponent"])
+      ({ desc: "notchHeight _ footer", ...χ.NamedArgsMarker }));
+    𝛄;
+    {
+      const [] = 𝛄.blockParams["default"];
+      {
+  // ...
+```
```

Comment:
Using these Greek letters makes the code/documentation less readable.
Why deviate from what is normally used.